test/e2e/specs/dashboard/settings__site-visibility.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @group calypso-pr
+ * @group dashboard
+ * @group settings
+ */
+
+import {
+	TestAccount,
+	getTestAccountByFeature,
+	envToFeatureKey,
+	envVariables,
+	DashboardPage,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+
+declare const browser: Browser;
+
+/**
+ * Tests that a user can edit privacy settings on a site via the dashboard.
+ *
+ * This test verifies that site visibility (privacy) options can be changed
+ * between Public, Private and Coming Soon modes.
+ */
+describe( 'Dashboard: Site Visibility Settings', function () {
+	let page: Page;
+	let dashboardPage: DashboardPage;
+	const accountName = getTestAccountByFeature( envToFeatureKey( envVariables ) );
+	const testAccount = new TestAccount( accountName );
+	let siteSlug: string;
+
+	beforeAll( async function () {
+		page = await browser.newPage();
+		await testAccount.authenticate( page );
+		dashboardPage = new DashboardPage( page );
+
+		// We need to extract the site slug for our test
+		// This assumes the user has at least one site
+		const url = testAccount.getSiteURL();
+		const urlParts = url.split( '/' );
+		console.log( urlParts );
+		siteSlug = urlParts[ urlParts.length - 2 ]; // Get the last part of the URL which should be the site slug
```

Comment:
Nit: it's not the last part.

Reply:
Maybe we should add a `getSiteSlug()` helper function to `TestAccount` 🤔

Reply:
> Maybe we should add a getSiteSlug() helper function to TestAccount 🤔

Definitely worth considering if it becomes more common